lib/discourse_slack/slack.rb

```diff
@@ -216,5 +216,51 @@ def self.notify(id)
 
       responses
     end
+
+    def self.channels
+      return [] if SiteSetting.slack_access_token.empty?
+
+      @channels = Rails.cache.fetch("slack_channels", expires_in: 15.minutes) do
+        http = Net::HTTP.new("slack.com" , 443)
+        http.use_ssl = true
+
+        uri = URI("https://slack.com/api/channels.list" +
+          "?token=#{ SiteSetting.slack_access_token }"
+        )
+
+        response = http.request(Net::HTTP::Get.new(uri))
+
+        if response
+          JSON.parse(response.body)["channels"]
+        else
+          []
+        end
+      end
+
+      @channels
+    end
+
+    def self.messages(channel_name, count)
+      return { "error": I18n.t('slack.errors.access_token_is_empty') } if SiteSetting.slack_access_token.empty?
+
+      http = Net::HTTP.new("slack.com" , 443)
+      http.use_ssl = true
+
+      channel = channels.find { |c| c["name"] == channel_name }
```

Comment:
Because [channels.history](https://api.slack.com/methods/channels.history) API method requires channel_id as argument. I guess if we know only channel name and not channel id then the only possible way to find channel id is [channels.list](https://api.slack.com/methods/channels.list) method by fetching all channels.